src/marqo/core/utils/vector_interpolation.py

```diff
@@ -69,23 +70,30 @@ def interpolate(self, vectors: List[List[float]], weights: List[float], prenorma
         if len(vectors) != len(weights):
             raise ValueError('Vectors and weights must have the same length')
 
-        weight_sum = sum(weights)
+        # Convert inputs to NumPy arrays for faster processing
+        np_vectors = np.array(vectors)
+        np_weights = np.array(weights)
+        
+        # Check if all vectors have the same length
+        if len(set(len(v) for v in vectors)) != 1:
+            raise ValueError('Vectors must have the same length')
+        
+        # Get sum of absolute values of weights
```

Comment:
This will go to 0 if and only if all weights are 0. Should we actually raise 400 for a 0 weight as it is meaningless?

Reply:
Updated error message, added AllZeroWeightsError. This should still raise a 400, for all 0 weights